components/in_memory_engine/src/write_batch.rs

```diff
@@ -472,7 +485,20 @@ impl WriteBatch for RegionCacheWriteBatch {
     }
 
     fn prepare_for_region(&mut self, region: &metapb::Region) {
+        if let Some(current_region) = &self.current_region
+            && current_region.id == region.id
+        {
+            return;
+        }
```

Comment:
Please add some comments about why the check is needed.

Reply:
Done